ipa-core/src/protocol/dp/mod.rs

```diff
@@ -98,6 +100,62 @@ where
     // Step 5 Transpose output representation
     Ok(Vec::transposed_from(&histogram_noised)?)
 }
+
+// dp_for_aggregation is currently where the DP parameters epsilon, delta
+// are introduced and then from those the parameters of the noise distribution to generate are
+// calculated for use in aggregating histograms.  The DP parameters query_epsilon and
+// per_user_credit_cap come as inputs to the query with per_user_sensitivity_cap = 2^{SS_BITS}
+/// # Errors
+/// will propogate errors from `apply_dp_noise`
+/// # Panics
+/// may panic from asserts down in  `gen_binomial_noise`
+/// may panic if running with DP noise but epsilon is not in the range (0,10].
+pub async fn dp_for_histogram<C, const B: usize, OV, const SS_BITS: usize>(
+    ctx: C,
+    histogram_bin_values: BitDecomposed<Replicated<Boolean, B>>,
+    dp_params: DPParams,
+) -> Result<Vec<Replicated<OV>>, Error>
+where
+    C: Context,
+    Boolean: Vectorizable<B> + FieldSimd<B>,
+    BitDecomposed<Replicated<Boolean, B>>: FromPrss<usize>,
+    OV: BooleanArray + U128Conversions,
+    Replicated<Boolean, B>: BooleanProtocols<C, B>,
+    Vec<Replicated<OV>>:
+        for<'a> TransposeFrom<&'a BitDecomposed<Replicated<Boolean, B>>, Error = LengthError>,
+{
+    match dp_params {
+        DPParams::TestingWithNoDP => Ok(Vec::transposed_from(&histogram_bin_values)?),
+        DPParams::WithDP(query_epsilon) => {
+            assert!(query_epsilon > 0.0 && query_epsilon <= 10.0);
+            let epsilon = query_epsilon;
+            let delta = 1e-6;
+            let success_prob = 0.5;
+            let dimensions = 1.0;
+            let quantization_scale = 1.0;
+            let per_user_credit_cap = 2_f64.powi(i32::try_from(SS_BITS).unwrap());
+            let ell_1_sensitivity = per_user_credit_cap;
+            let ell_2_sensitivity = per_user_credit_cap;
+            let ell_infty_sensitivity = per_user_credit_cap;
+            let num_bernoulli = find_smallest_num_bernoulli(
```

Comment:
I think it would be cleaner if you had a struct that contained all these parameters. This would get rid of `#[allow(clippy::too_many_arguments)]` in `find_smallest_num_bernoulli`.

You could also store the constants for delta, sucess_prob and such as part of that same struct or module so that if you want to provide some default implementations, it can be done as part of a new/default function and such.

Reply:
Thanks, that's a good suggestion. I agree there is still more we could do to clean up passing around these parameters.  I think I'd like to clean things up with changes like that in a later PR and do after we'd incorporated the noise for `feature_label_dot_product` as after that things will be more settled and it can be clearer what can be defaults in what cases.

Reply:
updated with a `NoiseParams` struct to hold all these internal parameters.